### ISSUE TITLE
Train IJEPA using proposed masking method

### ISSUE THREAD

**Issue by kutayeroglu** [kutayeroglu/ijepa#21]
- [x] Proposed masking method should work without errors
- [x] Same hyperparams w/multi-block should result in similar if not better loss values

Following these steps, we can start optimizing the training procedures

**Comment by kutayeroglu** [comment]
### Params to use for first experiment 

**Summary**
(ebs: 64, tf:%0.1, final_lr=1e-06, ema=0.999875)

    'optimization': {   'ema': [0.999875, 1.0],
                        'epochs': 300,
                        'final_lr': 1e-06,
                        'final_weight_decay': 0.4,
                        'ipe_scale': 1.0,
                        'lr': 3.1e-05,
                        'start_lr': 0.0002,
                        'warmup': 40,
                        'weight_decay': 0.04}}

**Detailed**
```
{   'data': {   'batch_size': 64,
                'color_jitter_strength': 0.0,
                'crop_scale': [0.3, 1.0],
                'crop_size': 224,
                'image_folder': '',
                'num_workers': 4,
                'pin_mem': True,
                'root_path': '/datasets/imagenet-object-localization-challenge/ILSVRC/Data/CLS-LOC',
                'train_fraction': 0.001,
                'use_color_distortion': False,
                'use_gaussian_blur': False,
                'use_horizontal_flip': False},
    'logging': {   'folder': '/users/kutay.eroglu/logs/ijepa/pretraining',
                   'write_tag': 'jepa'},
    'mask': {   'allow_overlap': False,
                'aspect_ratio': [0.75, 1.5],
                'enc_mask_scale': [0.85, 1.0],
                'min_keep': 10,
                'num_enc_masks': 1,
                'num_pred_masks': 4,
                'patch_size': 14,
                'pred_mask_scale': [0.15, 0.2]},
    'meta': {   'copy_data': False,
                'load_checkpoint': False,
                'model_name': 'vit_huge',
                'pred_depth': 12,
                'pred_emb_dim': 384,
                'read_checkpoint': None,
                'use_bfloat16': True},
    'optimization': {   'ema': [0.999875, 1.0],
                        'epochs': 300,
                        'final_lr': 1e-06,
                        'final_weight_decay': 0.4,
                        'ipe_scale': 1.0,
                        'lr': 3.1e-05,
                        'start_lr': 0.0002,
                        'warmup': 40,
                        'weight_decay': 0.04}}
```
